packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -363,6 +363,9 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
       final double oldPixels = pixels;
       _pixels = newPixels - overscroll;
       if (_pixels != oldPixels) {
+        if (outOfRange) {
```

Comment:
@Piinks Surprisingly this is enough, I cannot do `context.setIgnorePointer(!outOfRange)` as that breaks trackpad scrolling and one of the tests. By incorrectly ignoring when we should not be.